gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java

```diff
@@ -92,15 +91,21 @@
   public static final String GET_HIVE_TABLE = HIVE_REGISTER_METRICS_PREFIX + "getTableTimer";
   public static final String DROP_TABLE = HIVE_REGISTER_METRICS_PREFIX + "dropTableTimer";
   public static final String PATH_REGISTER_TIMER = HIVE_REGISTER_METRICS_PREFIX + "pathRegisterTimer";
+  public static final String OPTIMIZED_CHECK_ENABLED = "hiveRegister.doubleLockPatternCheck.enabled";
```

Comment:
Maybe `hiveRegister.cacheDbTableExistence`? Also, please add javadoc with the explanation of what this does.